packages/contracts/src/contracts/current/protocol/AssetProxy/proxies/ERC20Proxy.sol

```diff
@@ -33,16 +34,59 @@ contract ERC20Proxy is
 
     /// @dev Transfers ERC20 tokens. Either succeeds or throws.
     /// @param assetMetadata ERC20-encoded byte array.
-    /// @param from Address to transfer token from.
-    /// @param to Address to transfer token to.
-    /// @param amount Amount of token to transfer.
+    /// @param from Address to transfer asset from.
+    /// @param to Address to transfer asset to.
+    /// @param amount Amount of asset to transfer.
     function transferFrom(
         bytes assetMetadata,
         address from,
         address to,
         uint256 amount)
         external
         onlyAuthorized
+    {
+        transferFromInternal(
+            assetMetadata,
+            from,
+            to,
+            amount
+        );
+    }
+
+    /// @dev Makes multiple transfers of assets. Either succeeds or throws.
+    /// @param assetMetadata Array of byte arrays encoded for the respective asset proxy.
+    /// @param from Array of addresses to transfer assets from.
+    /// @param to Array of addresses to transfer assets to.
+    /// @param amounts Array of amounts of assets to transfer.
+    function batchTransferFrom(
+        bytes[] memory assetMetadata,
+        address[] memory from,
+        address[] memory to,
+        uint256[] memory amounts)
+        public
```

Comment:
We use external for `transferFrom` but public here, I expect it's the exchange contract who will be batchTransferring so should this also be external?

Reply:
External visibility doesn't work with ABIv2 unfortunately.